### PR TITLE
feat(specs): default authMode to WithinHeaders for Composition Client

### PR DESCRIPTION
**Pull request by e-krebs**
## 🧭 What and Why

🎟 JIRA Ticket: [EMERCH-1759](https://algolia.atlassian.net/browse/EMERCH-1759)

### Changes included:

- default authMode to WithinHeaders for Composition Client
- standardize variable name

## 🧪 Test

[EMERCH-1759]: https://algolia.atlassian.net/browse/EMERCH-1759?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/javascript/clients/client/builds/browser.mustache [hunk 1]
```diff
@@ -12,7 +12,7 @@
     logger: createNullLogger(),
     requester: createXhrRequester(),
     algoliaAgents: [{ segment: 'Browser' }],
-    authMode: 'WithinQueryParameters',
+    authMode: 'Within{{#isAlgoliaCompositionClient}}Headers{{/isAlgoliaCompositionClient}}{{^isAlgoliaCompositionClient}}QueryParameters{{/isAlgoliaCompositionClient}}',
```
Comment (shortcuts) [inline templates/javascript/clients/client/builds/browser.mustache:15]:
```suggestion
    authMode: 'Within{{#isCompositionClient}}Headers{{/isCompositionClient}}{{^isCompositionClient}}QueryParameters{{/isCompositionClient}}',
```

should be this as we generate each variable based on the spec name

Comment (e-krebs) [reply templates/javascript/clients/client/builds/browser.mustache:15]: I tested locally and the one I put is correct

Comment (shortcuts) [reply templates/javascript/clients/client/builds/browser.mustache:15]: ah nvm it's two different variables 🤔 would you mind updating https://github.com/algolia/api-clients-automation/blob/main/generators/src/main/java/com/algolia/codegen/AlgoliaJavascriptGenerator.java#L160to be:

```
    additionalProperties.put("is" + Helpers.capitalize(client) + "Client", true);
```

Comment (e-krebs) [reply templates/javascript/clients/client/builds/browser.mustache:15]: it's defined in `generators/src/main/java/com/algolia/codegen/AlgoliaJavascriptGenerator.java`

Comment (shortcuts) [reply templates/javascript/clients/client/builds/browser.mustache:15]: I can make the update actually if you want

Comment (shortcuts) [reply templates/javascript/clients/client/builds/browser.mustache]: I've pushed the update for it, we use this naming in all of our templates so it's mostly for consistency, sorry for the bad pointers!

**Review by shortcuts — APPROVED** [review]

**Pull request merged**